File: Cpp/simpson.cpp
ld simpson(ld a, ld b, vector<ld> v){
        ld h = (b - a) / maxn;
        ld s = evaluate(v, a) + evaluate(v, b);
        for(int i = 1; i <= maxn - 1; i++){
                ld h1 = a + h * i;
                s += evaluate(v, h1) * ((i & 1) ? 4 : 2);
        }
        return s * h / 3.0;
}

File: Python/simpson.py
def simpson(v, a, b):
        h = (b - a) / maxn
        s = evaluate(v, a) + evaluate(v, b)
        for i in range(1, maxn):
                h1 = a + h * i
                s += evaluate(v, h1) * (4 if i % 2 == 1 else 2)
        return s * h / 3.0
